File: app/src/test/java/igrek/songbook/dagger/base/BaseDaggerTest.java
package igrek.songbook.dagger.base;

import android.app.Activity;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;

import javax.inject.Inject;

import igrek.songbook.activity.MainActivity;
import igrek.songbook.dagger.DaggerIoc;
import igrek.songbook.logger.Logger;

@RunWith(RobolectricTestRunner.class)
public abstract class BaseDaggerTest {
	
	@Inject
	protected Activity activity;
	
	@Inject
	protected Logger logger;
	
	@Before
	public void setUp() {
		MainActivity activity = Robolectric.setupActivity(MainActivity.class);
		
		TestComponent component = DaggerTestComponent.builder()
				.factoryModule(new TestModule(activity))
				.build();
		
		DaggerIoc.setFactoryComponent(component);
		
		injectThis(component);
	}
	
	protected void injectThis(TestComponent component) {
		component.inject(this);
	}
}

File: app/src/test/java/igrek/songbook/dagger/base/BaseDaggerTest.kt
package igrek.songbook.dagger.base

import android.app.Activity
import igrek.songbook.activity.MainActivity
import igrek.songbook.dagger.DaggerIoc
import igrek.songbook.info.logger.Logger
import org.junit.Before
import org.mockito.Mockito
import javax.inject.Inject

abstract class BaseDaggerTest {

    @Inject
    lateinit var activity: Activity

    @Inject
    lateinit var logger: Logger

    @Before
    fun setUp() {
        val activity = Mockito.mock(MainActivity::class.java)

        val component = DaggerTestComponent.builder()
                .factoryModule(TestModule(activity))
                .build()

        DaggerIoc.factoryComponent = component

        injectThis(component)
    }

    private fun injectThis(component: TestComponent) {
        component.inject(this)
    }
}
